rali/rali_pybind/example/tf_mnistTrainingExample/tf_mnist_classification_rali.py

```diff
@@ -0,0 +1,192 @@
+
+	# setup keep_prob
+	#keep_prob = tf.placeholder(tf.float32)
+	input_X = tf.placeholder('float32',shape = (batch_size,784))
+	labels = tf.placeholder('float32',shape = (batch_size,10))
+	logits = neural_net(input_X)
+	cost = tf.reduce_mean( tf.nn.softmax_cross_entropy_with_logits(labels=labels, logits=logits), name="loss" )
+	optimizer = tf.train.AdamOptimizer().minimize(cost)
+	train_prediction = tf.nn.softmax(logits)
+	accuracy = compute_accuracy(train_prediction, labels)
+	correct_label = tf.argmax(labels, 1)
+	num_epochs = 10
+	crop_size = 28
+	TFRecordReaderType = 0
+	featureKeyMap = {
+		'image/encoded':'image_raw',
+		'image/class/label':'label',
+		'image/filename':''
+	}
+
+	trainPipe = HybridPipe(feature_key_map=featureKeyMap, tfrecordreader_type=TFRecordReaderType, batch_size=batch_size, num_threads=1, device_id=0, data_dir=data_path+"/train", crop=crop_size, rali_cpu=_rali_cpu)
+	valPipe = HybridPipe(feature_key_map=featureKeyMap, tfrecordreader_type=TFRecordReaderType, batch_size=batch_size, num_threads=1, device_id=0, data_dir=data_path+"/val", crop=crop_size, rali_cpu=_rali_cpu)
+	trainPipe.build()
+	valPipe.build()
+	trainIterator = RALIIterator(trainPipe)
+	valIterator = RALIIterator(valPipe)
+
+	with tf.Session() as sess:
+		sess.run(tf.initialize_all_variables())
+		for epoch in range(num_epochs):
+			print('\n\n----------------------------Training Model for Epoch: ', epoch, "-----------------------------------------------")
+			epoch_loss = 0
+			train_accuracy = 0
+			#for _ in range(int(mnist.train.num_examples/batch_size)):
+			for i, (image_train, label_train) in enumerate(trainIterator, 0):
+				image_train_res = image_train.reshape(batch_size, 784)
+				if 0:
+					fname = 'img_rali' + str(i)
+					np.save(fname, image_train_res, allow_pickle=False)
+					#	print(label_train)
+				train_label_one_hot_list = get_label_one_hot(label_train)
+				_, c, tacc = sess.run([optimizer, cost, accuracy], feed_dict={input_X:image_train_res, labels: train_label_one_hot_list})
+				epoch_loss += c
+				train_accuracy += tacc
+
+			print('Epoch', epoch, 'completed out of',num_epochs,'loss:',epoch_loss, 'accuracy:',(train_accuracy*100)/i, 'count :', i)
+			#run evaluation for every epoch
+			mean_acc = 0
+			#mean_loss = 0
+			print("\n\n----------------------------Evaluating Model ---------------------------------------------------------------")
+			for j, (val_image_ndArray, val_label_ndArray) in enumerate(valIterator, 0):
+				#val_image_ndArray_transposed = np.transpose(val_image_ndArray, [0, 2, 3, 1])
+				val_image_ndArray_res = val_image_ndArray.reshape(batch_size, 784)
+				val_label_one_hot_list = get_label_one_hot(val_label_ndArray)
+				val_accuracy = sess.run(accuracy,
+					#[optimizer, accuracy, prediction, correct_label, correct_pred],
+					feed_dict={input_X: val_image_ndArray_res, labels: val_label_one_hot_list})
+				mean_acc += val_accuracy
+				#mean_loss = mean_loss + val_loss
+				num_correct_predicate = 0
```

Comment:
Codacy found an issue: [local variable 'num_correct_predicate' is assigned to but never used (F841)](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6366586)